tests/Processor/SanitizeDataProcessorTest.php

```diff
@@ -12,181 +12,199 @@
+            }
 
-        $this->assertEquals(SanitizeDataProcessor::STRING_MASK, $data['extra']['ccnumba']);
-    }
+            unset($exceptionValue);
 
-    public function testSettingProcessorOptions()
-    {
-        $client = ClientBuilder::create()->getClient();
-        $processor = new SanitizeDataProcessor($client);
+            $event = $event->withException($inputData['exception']);
+        }
 
-        $this->assertEquals($processor->getFieldsRe(), '/(authorization|password|passwd|secret|password_confirmation|card_number|auth_pw)/i', 'got default fields');
-        $this->assertEquals($processor->getValuesRe(), '/^(?:\d[ -]*?){13,16}$/', 'got default values');
+        $event = $this->processor->process($event);
 
-        $options = [
-            'fields_re' => '/(api_token)/i',
-            'values_re' => '/^(?:\d[ -]*?){15,16}$/',
-        ];
+        if (isset($expectedData['request'])) {
+            $this->assertArraySubset($expectedData['request'], $event->getRequest());
+        }
 
-        $processor->setProcessorOptions($options);
+        if (isset($expectedData['extra_context'])) {
+            $this->assertArraySubset($expectedData['extra_context'], $event->getExtraContext());
+        }
 
-        $this->assertEquals($processor->getFieldsRe(), '/(api_token)/i', 'overwrote fields');
-        $this->assertEquals($processor->getValuesRe(), '/^(?:\d[ -]*?){15,16}$/', 'overwrote values');
-    }
+        if (isset($expectedData['exception'])) {
+            // We must convert the backtrace to a Stacktrace instance here because
+            // PHPUnit executes the data provider before the setUp method and so
+            // the client instance cannot be accessed from there
+            foreach ($expectedData['exception']['values'] as &$exceptionValue) {
+                $exceptionValue['stacktrace'] = Stacktrace::createFromBacktrace($this->client, $exceptionValue['stacktrace'], 'foo', 1);
+            }
 
-    /**
-     * @dataProvider overrideDataProvider
-     */
-    public function testOverrideOptions($processorOptions, $clientOptions)
-    {
-        $client = ClientBuilder::create($clientOptions)->getClient();
+            unset($exceptionValue);
 
-        /** @var SanitizeDataProcessor $processor */
-        $processor = $this->getObjectAttribute($client, 'processors')[0];
+            $this->assertArraySubset($expectedData['exception'], $event->getException());
+        }
 
-        $this->assertInstanceOf(SanitizeDataProcessor::class, $processor);
-        $this->assertEquals($processor->getFieldsRe(), $processorOptions[SanitizeDataProcessor::class]['fields_re'], 'overwrote fields');
-        $this->assertEquals($processor->getValuesRe(), $processorOptions[SanitizeDataProcessor::class]['values_re'], 'overwrote values');
+        $this->markTestIncomplete('Array scrubbing has not been implemented yet.');
```

Comment:
What kind of scrubbing is needed here?

Reply:
Like #212 (I believe), been looking at this recently.

Reply:
The test was incomplete even in `1.x` versions, from what I saw if an array like the one below is passed to the processor the `key_to_sanitize` item does not gets sanitized

```php
[
    'key_to_sanitize' => [
        'foo' => 'bar',
    ],
]
```